hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/FourToFiveUpgradeHandler.java

```diff
@@ -71,4 +83,26 @@ public Map<ConfigProperty, String> upgrade(HoodieWriteConfig config, HoodieEngin
     }
     return new HashMap<>();
   }
+
+  private String generatorPartitionPath(HoodieWriteConfig config, HoodieTableMetaClient metaClient) {
+    HoodieTableConfig tableConfig = metaClient.getTableConfig();
+    String partitionPath = DEPRECATED_DEFAULT_PARTITION_PATH;
+    try {
+      if (tableConfig.getPartitionFields().isPresent()) {
```

Comment:
Hive style partitioning is a table config, you can check that first, the file path pattern for hive style partioning is kind of deterministic, can we concatenate the path manually to get rid of the key gen and mock record?

Reply:
ok, I will update it. And I found one scene that was hard to fix, 
```shell
0.dt=20210101/hour=01
1.dt=20210101/hour=default
2.dt=default/hour=default
3.dt=20210102/hour=default
```
It is difficult to identify the default partition in row 1 and row 3
may be I can only check partitionPath: dt=default and  dt=default/ht=default

Reply:
> may be I can only check partitionPath: dt=default and dt=default/ht=default in this pr

Agree